warp10/src/main/java/io/warp10/script/functions/TOGTS.java

```diff
@@ -211,24 +236,22 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
             gts.setType(GeoTimeSerie.TYPE.BOOLEAN);
           }
         }
-        while (decoder.next()) {
+        if (decoder.next()) {
           Object value = decoder.getBinaryValue();
-          GTSHelper.setValue(gts, decoder.getTimestamp(), decoder.getLocation(), decoder.getElevation(), value, false);
-          // Test here the value type, because GTS do not handle BINARY type.
-          if (mustGuessTypeFromFirstValue) {
-            if (value instanceof String) {
-              enforcedType = "STRING";
-            } else if (value instanceof Boolean) {
-              enforcedType = "BOOLEAN";
-            } else if (value instanceof Long) {
-              enforcedType = "LONG";
-            } else if (value instanceof Double || value instanceof BigDecimal) {
-              enforcedType = "DOUBLE";
-            } else if (value instanceof byte[]) {
-              enforcedType = "BINARY";
-            }
-            mustGuessTypeFromFirstValue = false;
+          if (value instanceof String) {
+            enforcedType = "STRING";
+          } else if (value instanceof Boolean) {
+            enforcedType = "BOOLEAN";
+          } else if (value instanceof Long) {
+            enforcedType = "LONG";
+          } else if (value instanceof Double || value instanceof BigDecimal) {
+            enforcedType = "DOUBLE";
+          } else if (value instanceof byte[]) {
+            enforcedType = "BINARY";
           }
```

Comment:
must keep the `if (mustGuessTypeFromFirstValue) {` (it is here for the extra label to be correct)

```
          if(mustGuessTypeFromFirstValue) {
            if (value instanceof String) {
              enforcedType = "STRING";
            } else if (value instanceof Boolean) {
              enforcedType = "BOOLEAN";
            } else if (value instanceof Long) {
              enforcedType = "LONG";
            } else if (value instanceof Double || value instanceof BigDecimal) {
              enforcedType = "DOUBLE";
            } else if (value instanceof byte[]) {
              enforcedType = "BINARY";
            }
          }
```

Reply:
better explaination : 
the extra label needs to be correct when the filters did not enforce a type before. But if filter does enforce the type, the first value from encoder must not change the label.